### PR TITLE
Force logout users when their permissions change

### PR DESCRIPTION
**Pull request by robmarch2**

### CODE REVIEW

**Comment by robmarch2** [comment]
Still waiting on confirmation on whether or not we want to do this when permissions are added to the user as well - leaving as draft until then

File: datavault-common/src/main/java/org/datavaultplatform/common/util/RoleUtils.java [hunk 1]
```diff
@@ -12,4 +19,15 @@ private RoleUtils() {}
     public static boolean isReservedRoleName(String roleName) {
         return IS_ADMIN_ROLE_NAME.equalsIgnoreCase(roleName) || DATA_OWNER_ROLE_NAME.equalsIgnoreCase(roleName);
     }
+
+    public static boolean hasReducedPermissions(Collection<PermissionModel> originalPermissions,
+                                                Collection<PermissionModel> newPermissions) {
+        Set<Permission> original = originalPermissions.stream()
```
Comment (garyttierney) [inline datavault-common/src/main/java/org/datavaultplatform/common/util/RoleUtils.java]: We're only comparing the permissions of the roles that were updated, is that right?

Comment (robmarch2) [reply datavault-common/src/main/java/org/datavaultplatform/common/util/RoleUtils.java]: It was right, but not any more. This change originally only cared if permissions were downgraded so we only needed to check the updates here, but we got confirmation yesterday that we also want to force logout any users that gain new permissions so this bit of code will be gotten rid of shortly

Comment (garyttierney) [reply datavault-common/src/main/java/org/datavaultplatform/common/util/RoleUtils.java]: Ah, nice. Makes it simpler.

**Review by garyttierney — APPROVED** [review]

**Pull request merged**